kernel/task/src/lib.rs

```diff
@@ -1181,88 +1159,152 @@ impl Drop for JoinableTaskRef {
+    /// This is only usable within task cleanup functions to indicate
     /// that the current task has cleanly exited.
     /// 
-    /// # Locking / Deadlock
-    /// This method obtains a writable lock on the underlying Task's inner state.
-    /// 
     /// # Return
     /// * Returns `Ok` if the exit status was successfully set.     
-    /// * Returns `Err` if this `Task` was already exited, and does not overwrite the existing exit status. 
+    /// * Returns `Err` if this `Task` was already exited;
+    ///   the existing exit status will not be overwritten.
     ///  
     /// # Note 
     /// The `Task` will not be halted immediately -- 
     /// it will finish running its current timeslice, and then never be run again.
-    #[doc(hidden)]
     pub fn mark_as_exited(&self, exit_value: Box<dyn Any + Send>) -> Result<(), &'static str> {
         self.internal_exit(ExitValue::Completed(exit_value))
     }
 
     /// Call this function to indicate that this task has been cleaned up (e.g., by unwinding)
     /// and it is ready to be marked as killed, i.e., it will never run again.
-    /// This task (`self`) must be the currently executing task, 
-    /// you cannot invoke `mark_as_killed()` on a different task.
     /// 
-    /// If you want to kill another task, use the [`kill()`](method.kill) method instead.
+    /// If you want to kill another task, use [`TaskRef::kill()`] instead.
     /// 
-    /// This should only be used within task cleanup functions (e.g., after unwinding) to indicate
+    /// This is only usable within task cleanup functions (e.g., after unwinding) to indicate
     /// that the current task has crashed or failed and has been killed by the system.
     /// 
-    /// # Locking / Deadlock
-    /// This method obtains a writable lock on the underlying Task's inner state.
-    /// 
     /// # Return
     /// * Returns `Ok` if the exit status was successfully set.     
     /// * Returns `Err` if this `Task` was already exited, and does not overwrite the existing exit status. 
     ///  
     /// # Note 
     /// The `Task` will not be halted immediately -- 
     /// it will finish running its current timeslice, and then never be run again.
-    #[doc(hidden)]
     pub fn mark_as_killed(&self, reason: KillReason) -> Result<(), &'static str> {
-        if with_current_task(|t| t == self)
-            .unwrap_or(false)
-        {
-            self.internal_exit(ExitValue::Killed(reason))
-        } else {
-            Err("`mark_as_exited()` can only be invoked on the current task, not on another task.")
+        self.internal_exit(ExitValue::Killed(reason))
+    }
+
+    /// Reaps this task (if orphaned) by taking and dropping its exit value and removing it
+    /// from the system task list.
+    ///
+    /// If this task has *not* been orphaned, meaning it is still joinable,
+    /// then this function does nothing.
+    pub fn reap_if_orphaned(&self) {
```

Comment:
This should technically be implemented for `Task`, not `ExitableTaskRef` since any task should be able to reap any other task.

Reply:
Hmm, I don't necessarily agree with that. The point of the new `ExitableTaskRef` type is to ensure that only a task itself can clean itself up in certain ways (e.g., marking itself as exited to indicate a regular return). I'd argue that if the task has been orphaned (it's `JoinableTaskRef` has been dropped elsewhere) then it should be the one to clean itself up, not any other random task.

Reply:
If we have a future use case that requires a random foreign task to reap another random task, then we can move it.